From 404a309ea7de7f28cd2805cf4712ee40eb6d0bba Mon Sep 17 00:00:00 2001
From: Wenzong Fan <wenzong.fan@windriver.com>
Date: Tue, 23 Feb 2016 03:59:34 -0500
Subject: [PATCH] refpolicy-rbac: fix install errors

Fix install errors:

| Re-declaration of role engineer_r
| Failed to create node
| Bad role declaration at line 1 of .../tmp/work/qemux86_64-wrs-linux/refpolicy-rbac/2.20151208-r0/image/var/lib/selinux/wr-rbac/tmp/modules/100/hierarchrole/cil
| Failed to build ast
| semodule:  Failed!

Upstream-Status: Inappropriate [no upstream]

Signed-off-by: Wenzong Fan <wenzong.fan@windriver.com>
---
 policy/modules/roles/hierarchrole.te | 16 ++++++++++++----
 1 file changed, 12 insertions(+), 4 deletions(-)

diff --git a/policy/modules/roles/hierarchrole.te b/policy/modules/roles/hierarchrole.te
index bf675b8..933857c 100644
--- a/policy/modules/roles/hierarchrole.te
+++ b/policy/modules/roles/hierarchrole.te
@@ -5,7 +5,9 @@ policy_module(hierarchrole, 1.0.0)
 # Declare engineer role and type
 #
 
-role engineer_r;
+gen_require(`
+	role engineer_r;
+')
 userdom_restricted_user_template(engineer)
 
 # Chcat will access /selinux/mls whose type is security_t, the interface
@@ -22,7 +24,9 @@ seutil_run_newrole(engineer_t, engineer_r)
 # Declare senior engineer
 #
 
-role senior_engineer_r;
+gen_require(`
+	role senior_engineer_r;
+')
 userdom_restricted_user_template(senior_engineer)
 
 selinux_get_enforce_mode(senior_engineer_t)
@@ -46,7 +50,9 @@ allow senior_engineer_r engineer_r;
 # Declare manager
 #
 
-role manager_r;
+gen_require(`
+	role manager_r;
+')
 userdom_restricted_user_template(manager)
 
 selinux_get_enforce_mode(manager_t)
@@ -72,7 +78,9 @@ allow manager_r engineer_r;
 # Declare technical leader
 #
 
-role technical_lead_r ;
+gen_require(`
+	role technical_lead_r;
+')
 userdom_restricted_user_template(technical_lead)
 
 selinux_get_enforce_mode(technical_lead_t)
-- 
1.9.1

